### PR TITLE
Fixed rounding errors for JSON>YAML converter

### PR DESCRIPTION
**Pull request by sapvissers**
## Pull request type

Please check the type of change your PR introduces:

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Internationalization and localization
- [ ] Other (please describe):

## What is the current behavior?

Issue Number: #168 

## What is the new behavior?

- Added a custom `DoubleConverter` to handle the correct double conversion from JSON to YAML, which fixed all rounding errors.

## Other information

I didn't implement the `ReadYaml()` method in the `DoubleConverter` because there is no use for it.

## Quality check

Before creating this PR, have you:

- [x] Followed the code style guideline as described in [CONTRIBUTING.md](https://github.com/veler/DevToys/blob/main/CONTRIBUTING.md)
- [x] Verified that the change work in Release build configuration
- [x] Checked all unit tests pass

### CODE REVIEW

File: src/dev/impl/DevToys/ViewModels/Tools/Converters/JsonYaml/DoubleConverter.cs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    internal class DoubleConverter : IYamlTypeConverter
+    {
+        public bool Accepts(Type type)
+        {
+            return type == typeof(double);
+        }
+
+        public object ReadYaml(IParser parser, Type type)
+        {
+            throw new NotImplementedException();
```
Comment (veler) [inline src/dev/impl/DevToys/ViewModels/Tools/Converters/JsonYaml/DoubleConverter.cs:19]: What happens if the user converts from YAML to JSON? Would this method be called?

File: src/dev/DevToys.Startup/Package.appxmanifest [hunk 2]
```diff
@@ -15,7 +15,7 @@
     <Identity
         Name="DevToysPreview"
         Publisher="CN=etiennebaudoux"
-        Version="0.0.0.0" />
+        Version="1.0.1.0" />
```
Comment (veler) [inline src/dev/DevToys.Startup/Package.appxmanifest:18]: Please revert this change.

File: src/dev/shared/SharedAssemblyInfo.cs [hunk 3]
```diff
@@ -37,7 +37,7 @@
 
 // Please DO NOT commit changes on the 2 lines below unless
 // you're about to release a new version of the app on the Store.
-[assembly: AssemblyVersion("0.0.0.0")]
-[assembly: AssemblyFileVersion("0.0.0.0")]
+[assembly: AssemblyVersion("1.0.1.0")]
```
Comment (veler) [inline src/dev/shared/SharedAssemblyInfo.cs:40]: Please revert this change too.

**Review by veler — COMMENTED** [review]
Hello,

Thank you for making this fix. A couple of comments, but overall it looks good. :)

**Comment by veler** [comment]
Additionally, I feel like this PR could fix #202, isn't it? Mind giving it a quick try?

**Comment by akarboush** [comment]
Is there really a need for a custom converter?
Why just not specify that floating point numbers should be parsed as `decimal` in the [JsonConvert.DeserializeObject](https://github.com/veler/DevToys/blob/3b612270587f89479e191fdddcab18ca4d08d501/src/dev/impl/DevToys/ViewModels/Tools/Converters/JsonYaml/JsonYamlToolViewModel.cs#L233)
```csharp
private readonly Newtonsoft.Json.JsonSerializerSettings _defaultJsonSerializerSettings = new () 
{
   FloatParseHandling = FloatParseHandling.Decimal 
};
..........
dynamic? jsonObject = JsonConvert.DeserializeObject<ExpandoObject>(input, _defaultJsonSerializerSettings );

```
Decimal would also handle a larger number than double

**Comment by sapvissers** [comment]
> Is there really a need for a custom converter? Why just not specify that floating point numbers should be parsed as `decimal` in the [JsonConvert.DeserializeObject](https://github.com/veler/DevToys/blob/3b612270587f89479e191fdddcab18ca4d08d501/src/dev/impl/DevToys/ViewModels/Tools/Converters/JsonYaml/JsonYamlToolViewModel.cs#L233)
> 
> ```cs
> private readonly Newtonsoft.Json.JsonSerializerSettings _defaultJsonSerializerSettings = new () 
> {
>    FloatParseHandling = FloatParseHandling.Decimal 
> };
> ..........
> dynamic? jsonObject = JsonConvert.DeserializeObject<ExpandoObject>(input, _defaultJsonSerializerSettings );
> ```
> 
> Decimal would also handle a larger number than double

Thank you for this feedback, didn't know this was possible!
I've refactored the code as you sugested and it works great.

**Comment by sapvissers** [comment]
> Additionally, I feel like this PR could fix #202, isn't it? Mind giving it a quick try?

This PR doesn't fix #202, I did look at it briefly, but couldn't find a quick solution so I'll look into that issue when I have the time.

**Comment by veler** [comment]
That looks great! Thank you very much for fixing this 🚀😁
